### DISCUSSION TITLE
VideoTexture to Skybox?

### DISCUSSION THREAD

**Discussion by JeromeCHA** [sceneview/sceneview#385] [Q&A]
I am trying to put a videoTexture to a skybox to have a 360 panorama video. 

I followed [VideoMaterial](https://github.com/SceneView/sceneview-android/blob/c3b8c5a0fca0dbb43c971a91fe808456f51cd65e/sceneview/src/main/java/io/github/sceneview/material/VideoMaterial.kt) code source, and I was able to get exoplayer's texture and map it into a node. 

However, I would like to display it as a skybox.

When I tried, by doing

```kotlin
                   val texture = VideoTexture.Builder()
                        .stream(stream)
                        .build(engine)

                    skybox = Skybox.Builder()
                        .environment(texture)
                        .build(engine)
```
I got the following error.

```
Filament                 E  Panic
                            in build:82
                            reason: environment maps must be a cubemap
Filament                 E  
libc                     A  Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 21944 (sceneviewsample), pid 21944 (sceneviewsample)
```

Is there any way to get a cubemap from a videoTexture?
I tried with `IBLPrefilter`, but it doesn't work because VideoTexture is not a 2D texture. It is quite confusing even looking at the code.

https://github.com/SceneView/sceneview-android/blob/ad5237503650d15bf780353aa8de159b7a08c295/sceneview_2_0_0/src/main/java/io/github/sceneview/utils/IBLPrefilter.kt#L32-L48

Can someone please tell me what is the best way to do it?

**Comment by ThomasGorisse** [comment]
Mapping a video to a skybox (360-degree video) is not directly supported as a high-level API in v4.0.0-rc.1. However, you could achieve this by:

1. **VideoNode as a large sphere** — Create a large inverted sphere with a video texture, place the camera inside
2. **Custom Filament material** — Create a skybox material that reads from an `ExternalTexture` (video source)

For 360 video playback specifically, consider using ExoPlayer with a spherical renderer, or combine SceneView's `VideoNode` with a sphere geometry. This is an advanced use case that requires working with Filament materials directly.
